man/kpkg.8
.\" Automatically generated by Pandoc 3.1.3
.\"
.\" Define V font for inline verbatim, using C font in formats
.\" that render this, and otherwise B font.
.ie "\f[CB]x\f[]"x" \{\
. ftr V B
. ftr VI BI
. ftr VB B
. ftr VBI BI
.\}
.el \{\
. ftr V CR
. ftr VI CI
. ftr VB CB
. ftr VBI CBI
.\}
.TH "kpkg" "8" "" "" ""
.hy
.SH NAME
.PP
kpkg - package manager
.SH SYNOPSIS
.PP
\f[B]kpkg\f[R] [h] [b] [ins] [inf] [r] [upd] [upg] [PACKAGES] [-y] [-o]
[-r] [ROOTFS]
.SH DESCRIPTION
.PP
kpkg is a package manager written in Nim.
.PP
kpkg uses simple build scripts called runfiles, similar to PKGBUILDs
found in the pacman package manager.
You can learn more about runfiles in kpkg_run(8).
.SH COMMANDS
.PP
\f[B]h\f[R] Output help
.PP
\f[B]b\f[R] build package
.PP
\f[B]ins\f[R] install binary package
.PP
\f[B]inf\f[R] get info about a package
.PP
\f[B]r\f[R] remove package
.PP
\f[B]upd\f[R] update repository
.PP
\f[B]upg\f[R] upgrade packages
.PP
\f[B]v\f[R] print version
.PP
\f[B]-o\f[R] only build, dont install
.PP
\f[B]-r\f[R] Rootfs argument.
.PP
\f[B]-y\f[R] Auto-accept installation.
.PP
For full arguments, please see \f[V]kpkg help\f[R]
.SH ENVIRONMENT VARIABLES
.IP \[bu] 2
\f[V]KPKG_ENABLE_DEBUG\f[R]: Boolean that enables debug messages on
kpkg.
Is very useful for bug reports.
Set to one of \f[V]y\f[R], \f[V]yes\f[R], \f[V]true\f[R] or \f[V]1\f[R]
to enable.
.SH AUTHOR
.PP
Written by Kreato.
.SH COPYRIGHT
.PP
kpkg is free software: you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free
Software Foundation, either version 3 of the License, or (at your
option) any later version.
.PP
kpkg is distributed in the hope that it will be useful, but WITHOUT ANY
WARRANTY; without even the implied warranty of MERCHANTABILITY or
FITNESS FOR A PARTICULAR PURPOSE.
See the GNU General Public License for more details.
.PP
You should have received a copy of the GNU General Public License along
with kpkg.
If not, see <https://www.gnu.org/licenses/>.


kpkg/kpkg.nim
# kpkg - Simple, efficient and fast package manager
# Copyright 2022 Kreato
#
# This file is part of Kreato Linux.
#
# Kreato Linux is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# Kreato Linux is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with Kreato Linux.  If not, see <https://www.gnu.org/licenses/>.

import cligen
import commands/getsetcmd
import commands/infocmd
import commands/buildcmd
import commands/updatecmd
import commands/removecmd
import commands/upgradecmd
import commands/installcmd
import commands/searchcmd
import commands/cleancmd
import commands/auditcmd
import commands/providescmd
import commands/checkcmd
import commands/listcmd
import commands/initcmd
import ../common/version

if commitVer != "unavailable":
  clCfg.version = "kpkg "&ver&", commit "&commitVer
else:
  clCfg.version = "kpkg "&ver


dispatchMultiGen([ "init" ], [sandbox, mergeNames = @[ "kpkg", "init" ]], [package, mergeNames = @["kpkg", "init"]], #[ insert system here ]# [override, mergeNames = @["kpkg", "init"]])

dispatchMulti(
  [
  build, help = {
    "packages": "The package names",
    "root": "The directory the package is gonna be installed to",
    "yes": "Automatically say 'yes' to every question",
    "no": "Automatically say 'no' to every question",
    "useCacheIfAvailable": "Uses cached build (if available)",
    "dontInstall": "Only build, don't install",
    "tests": "Enable/Disable Tests on packages",
    "forceInstallAll": "Force reinstall every dependency",
    "isInstallDir": "Build package from specified path",
    "ignorePostInstall": "Ignore if postInstall fails",
    "checkReproducibility": "Download the same package from binRepos and compares the checksums"
  },
    suppress = @["isUpgrade"] # Internal variable for commands/upgradecmd
  ],

  [
  install, help = {
    "root": "The directory the package is gonna be installed to",
    "yes": "Automatically say 'yes' to every question",
    "no": "Automatically say 'no' to every question",
    "offline": "Offline mode, errors out if tarball is attempted to get downloaded off binrepo",
    "downloadOnly": "Only download the binary, don't install",
    "ignoreDownloadErrors": "Ignore errors that may occur while downloading packages"
    },
    suppress = @["isUpgrade", "basePackage"] # Internal variable for commands/upgradecmd, Internal variable of kreastrap
  ],

  [
  search, help = {
    "colors": "Enable color output"
  }
  ],

  [
  remove, help = {
    "yes": "Automatically say 'yes' to every question",
    "root": "The directory the package is gonna be removed from",
    "force": "Ignore dependency checks",
    "autoRemove": "Remove unused dependencies",
    "configRemove": "Remove configuration"
    }
  ],

  [
  provides, help = {
    "files": "Files to search",
    "color": "Enable color output "
    }
  ],

  [
  update, help = {
    "repo": "The kpkg repository Git URL",
    "path": "The kpkg repository path",
    "branch": "The kpkg repository branch. Also supports commits."
    }
  ],

  [
  upgrade, help = {
    "root": "The directory the packages are gonna be upgraded on",
    "builddir": "Set a custom build directory",
    "yes": "Automatically say 'yes' to every question",
    "no": "Automatically say 'no' to every question",
    "srcdir": "Set a custom source directory"
    }
  ],

  [
  clean, help = {
    "sources": "Remove source tarballs from cache",
    "binaries": "Remove binary tarballs from cache",
    "cache": "Remove ccache directory",
    "environment": "Remove build environment directory"
  }
  ],
  
  [
  audit, help = {
    "description": "Show descriptions of CVEs",
    "fetch": "Fetch and build/install the vulnerability database",
    "fetchBinary": "Fetch already-built SQLite database"
    }
  ],

  [
  info, help = {
  "testing": "Don't error if package isn't installed"
  }
  ],
  [
  check, help = {
    "root": "The directory the packages are gonna be checked on",
    "package": "Set a specific package to check"
  }
  ],
  [
  list
  ],
  [
  init, doc = "Initialize multiple types of files", usage = "$doc\n", stopWords = @["sandbox", "package", "override", "hook"]
  ],
  [
  get, help = {
    "invocations": "The invocation(s) to read"
    }
  ],
  [
  getsetcmd.set, help = {
    "invocation": "The invocation to write",
    "file": "URL/Path to the file with invocations",
    "append": "Append to the value instead of replacing it"
    }
  ]
)
